### PR TITLE
Implemented a base channel

### PR DESCRIPTION
**Pull request by flmichel**
1. Implement a generic channel that can be use for any type of channel (Lao channel, election channel...).
2. Update the actual Lao channel by using the new generic channel (base channel).

### CODE REVIEW

File: be1-go/hub/organizer.go [hunk 1]
```diff
@@ -214,68 +213,26 @@ func (o *organizerHub) createLao(publish message.Publish) error {
...
 
 func (c *laoChannel) Subscribe(client *Client, msg message.Subscribe) error {
-	log.Printf("received a subscribe with id: %d", msg.ID)
-	c.clientsMu.Lock()
-	defer c.clientsMu.Unlock()
-
-	c.clients[client] = struct{}{}
-
-	return nil
+	return c.baseChannel.Subscribe(client, msg)
```
Comment (gnarula) [inline be1-go/hub/organizer.go:220]: The nice thing about embedding is the struct that embeds a parent struct as the ability to call methods on the embedded struct. Have a look at https://play.golang.org/p/iNTjT13SaIx for an example.

**Review by gnarula — CHANGES_REQUESTED** [review]
Thanks for the PR! I think it's fine if only baseChannel implements Subscribe, Unsubscribe and Catchup. The structs embedding it will automatically pass on the call to those methods - we don't need to define them again. Please have a look at the playground link I shared for an example.

File: be1-go/hub/organizer.go (hunk 1 above)
Comment (flmichel) [reply be1-go/hub/organizer.go]: Thank you, I removed the methods in laoChannel.

**Review by gnarula — APPROVED** [review]
LGTM!

**Pull request merged**